client/src/lib.rs

```diff
@@ -590,6 +593,132 @@ impl Namespace {
+    Duration::from_secs(60 * 60 * 24 * 3);
+
+/// Helper struct for `register_namespace`.
+#[derive(Clone, derive_builder::Builder)]
+pub struct RegisterNamespaceOptions {
+    /// Name (required)
+    #[builder(setter(into))]
+    pub namespace: String,
+    /// Description (required)
+    #[builder(setter(into))]
+    pub description: String,
+    /// Owner's email
+    #[builder(default)]
+    pub owner_email: String,
+    /// Workflow execution retention period
+    #[builder(default = "DEFAULT_WORKFLOW_EXECUTION_RETENTION_PERIOD")]
+    pub workflow_execution_retention_period: Duration,
+    /// Cluster settings
+    #[builder(setter(strip_option, custom), default)]
+    pub clusters: Vec<ClusterReplicationConfig>,
+    /// Active cluster name
+    #[builder(setter(into), default)]
+    pub active_cluster_name: String,
+    /// Custom Data
+    #[builder(default)]
+    pub data: HashMap<String, String>,
+    /// Security Token
+    #[builder(default)]
+    pub security_token: String,
+    /// Global namespace
+    #[builder(default)]
+    pub is_global_namespace: bool,
+    /// History Archival setting
+    #[builder(setter(into), default = "ArchivalState::Unspecified")]
+    pub history_archival_state: ArchivalState,
+    /// History Archival uri
+    #[builder(setter(strip_option), default)]
+    pub history_archival_uri: String,
+    /// Visibility Archival setting
+    #[builder(setter(into), default = "ArchivalState::Unspecified")]
+    pub visibility_archival_state: ArchivalState,
+    /// Visibility Archival uri
+    #[builder(setter(strip_option), default)]
+    pub visibility_archival_uri: String,
+}
+
+impl RegisterNamespaceOptions {
+    /// Builder convenience.  Less `use` imports
+    pub fn builder() -> RegisterNamespaceOptionsBuilder {
+        Default::default()
+    }
+}
+
+impl Into<RegisterNamespaceRequest> for RegisterNamespaceOptions {
+    fn into(self) -> RegisterNamespaceRequest {
+        RegisterNamespaceRequest {
+            namespace: self.namespace,
+            description: self.description,
+            owner_email: self.owner_email,
+            workflow_execution_retention_period: self
+                .workflow_execution_retention_period
+                .try_into()
+                .ok(),
+            clusters: self.clusters,
+            active_cluster_name: self.active_cluster_name,
+            data: self.data,
+            security_token: self.security_token,
+            is_global_namespace: self.is_global_namespace,
+            history_archival_state: self.history_archival_state as i32,
+            history_archival_uri: self.history_archival_uri,
+            visibility_archival_state: self.visibility_archival_state as i32,
+            visibility_archival_uri: self.visibility_archival_uri,
+        }
+    }
+}
+
+impl RegisterNamespaceOptionsBuilder {
+    /// Custum builder function for convenience
+    pub fn clusters(&mut self, clusters: Vec<String>) {
```

Comment:
maybe call this `cluster_names` - also worth noting in the docstring it could blow away any previously set cluster configs if that had been done already.